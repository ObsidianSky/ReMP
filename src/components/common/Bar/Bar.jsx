import React from 'react';
import PropTypes from 'prop-types';
import './Bar.scss'

const Bar = ({ title, children }) => (
    <div className="bar">
        <div className="bar__inner">
            <div className="bar__left">
                <div className="bar__title">{title}</div>
            </div>
            <div className="bar__right">
                {children}
            </div>
        </div>
    </div>
);

Bar.defaultProps = {
    title: ''
};

Bar.propTypes = {
    title: PropTypes.string,
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node
    ]),
};

export default Bar;