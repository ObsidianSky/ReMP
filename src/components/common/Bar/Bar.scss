@import '../../../assets/styles/variables';

.bar {
  background: $almost-white;
  padding: 18px 0;
  min-height: 56px;

  &__inner {
    max-width: 1200px;
    margin: 0 auto;
    display: flex;
    justify-content: space-between;
    align-items: center;
  }

  &__title {
    font-weight: 600;
    color: $text-black;
  }
}