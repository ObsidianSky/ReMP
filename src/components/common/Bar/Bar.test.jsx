import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Bar from './Bar';

describe('Bar', () => {
    let props;

    beforeEach(() => {
        props = {
            title: 'title',
            children: <div>children</div>
        }
    });

    test('should renders correctly', () => {
        const wrapper = shallow(<Bar />);

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

