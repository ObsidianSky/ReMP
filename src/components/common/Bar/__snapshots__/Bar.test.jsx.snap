// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar should renders correctly 1`] = `
<div
  className="bar"
>
  <div
    className="bar__inner"
  >
    <div
      className="bar__left"
    >
      <div
        className="bar__title"
      />
    </div>
    <div
      className="bar__right"
    />
  </div>
</div>
`;
